docker/entrypoint.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/sh
+
+# InfluxDB configuration.
+INFLUXDB_HOST=${INFLUXDB_HOST:-influxdb}
+INFLUXDB_PORT=${INFLUXDB_PORT:-25826}
+
+# fritzcollectd plugin configuration.
+FRITZBOX_HOST=${FRITZBOX_HOST:-fritz.box}
+FRITZBOX_PORT=${FRITZBOX_PORT:-49000}
+FRITZBOX_USER=${FRITZBOX_USER:-dslf-config}
+FRITZBOX_HOSTNAME=${FRITZBOX_HOSTNAME:-fritzbox}
+VERBOSE=${VERBOSE:-false}
+
+cat > /etc/collectd/collectd.conf <<EOF
+FQDNLookup true
+LoadPlugin logfile
+
+<Plugin logfile>
+    LogLevel "info"
+    File STDOUT
+    Timestamp true
+    PrintSeverity false
+</Plugin>
+
+LoadPlugin network
+LoadPlugin python
+
+<Plugin network>
+    Server "$INFLUXDB_HOST" "$INFLUXDB_PORT"
+</Plugin>
+<Plugin python>
+    ModulePath "/fritzcollectd"
+    Import "fritzcollectd"
+    LogTraces true
+
+    <Module fritzcollectd>
+        Address "$FRITZBOX_HOST"
+        Port $FRITZBOX_PORT
+        User "$FRITZBOX_USER"
+        Password "$FRITZBOX_PASSWORD"
+        Hostname "$FRITZBOX_HOSTNAME"
+        Instance "$INSTANCE"
+        Verbose "$VERBOSE"
+    </Module>
+</Plugin>
+EOF
+
+# This container supports running with fritzcollectd's git repository mounted
+# to /fritzcollectd (and we then just install the dependencies).
+if [ -d "/fritzcollectd" ]; then
+    pip install -r/fritzcollectd/requirements.txt
+else
+    pip install fritzcollectd
```

Comment:
This should be done in the Dockerfile and not at runtime or you face a debugging nightmare and can't version pin or update the software easily or you download it with every start of the image which is als not great. This would be the main reason why I wouldn't use this Dockerimage and do my own.

Also this entrypoint blocks simply running sh as without changing the entrypoint which is annoying. A good and easy way to do this can be found here https://github.com/docker-library/redis/blob/6ec0ad5628df2404509f776e9c70fbecf5364c10/5.0/docker-entrypoint.sh.
This also drops the user to a non root user which is preferred in most cases.

Reply:
Also in the Dockerimage pip shoulnd't cache anything.
``pip install --no-cache-dir --progress-bar off``